casper/src/main/scala/coop/rchain/casper/EstimatorHelper.scala

```diff
@@ -151,7 +152,7 @@ object EstimatorHelper {
   ): Map[Blake2b256Hash, Set[TuplespaceEvent]] = {
     val nonPersistentProducesInComms = b.comms.flatMap(_.produces).filterNot(_.persistent)
     val nonPersistentConsumesInComms = b.comms.map(_.consume).filterNot(_.persistent)
-    val freeProduces                 = b.produces.diff(nonPersistentProducesInComms)
+    val freeProduces                 = b.produces.diff(nonPersistentProducesInComms).filter(_.sequenceNumber == 0)
```

Comment:
1. why did that have to change? How does addition of peeks affects `nonPersistentProducesInComms`? Is that name still correct? Or did we have a bug?
2. we shouldn't be relying on sequenceNumber, we intend to remove it after all.

Reply:
1) No, it's still correct. It affects `freeProduces`. Those with a `sequenceNumber > 0` were reintroduced into the tuplespace in the Reducer, but technically they were peeked, so they shouldn't count as "free"
2) There's no other way while we still have them. Once sequenceNumbers in produces are replaced with counters these will provide the necessary information

Reply:
Have you checked that?

Reply:
You're right, I'll tweak this part of code

Reply:
No it's a different mechanism, for details see https://github.com/rchain/rchain/pull/1817

Reply:
> There's no other way while we still have them.

:stuck_out_tongue_winking_eye: